api/app/zonal_stats.py

```diff
@@ -368,6 +376,48 @@ def intersect_percentage(masked) -> float:
     except ValueError as error:
         stats_results = []
 
+    # cleanup data and remove nan values
+    # add intersect stats if requested
+    clean_results = []
+    for result in stats_results:
+        stats_properties: dict = result if not geojson_out else result["properties"]
+
+        # clean results
+        clean_stats_properties = {
+            k: 0 if str(v).lower() == "nan" else v for k, v in stats_properties.items()
+        }
+
+        # calculate intersect_percentage
+        if intersect_comparison is not None:
+            safe_prefix = prefix or ""
+            total = (
+                float(clean_stats_properties[f"{safe_prefix}count"])
+                + clean_stats_properties[f"{safe_prefix}nodata"]
+            )
+            if total == 0:
+                intersect_percentage = Percentage(0.0)
+            else:
+                intersect_percentage = Percentage(
+                    clean_stats_properties[f"{safe_prefix}intersect_pixels"] / total
+                )
+
+            clean_stats_properties = {
+                **clean_stats_properties,
+                f"{safe_prefix}intersect_percentage": intersect_percentage,
+            }
+
+        # merge properties back at the proper level
+        # and filter out properties that have "no" intersection
+        # by setting a limit at 0.005 (0.5%).
```

Comment:
@wadhwamatic As it happened there were actually very few admin boundaries with `zero` value, but very small percentages. I set an arbitrary limit of 0.5%. Let me know what you think.